File: 0501-1000/0824_toGoatLatin.java
class Solution {
    public String toGoatLatin(String sentence) {
        String[] words = sentence.split(" ");
        StringBuilder converted = new StringBuilder();
        
        for (int i = 0; i < words.length; i++) {
            String word = words[i];
            String rep = repeatChar('a', i + 1);

            if (isVowel(Character.toLowerCase(word.charAt(0)))) {
                converted.append(word).append("ma").append(rep);
            } else {
                String second = word.substring(1);
                char first = word.charAt(0);
                converted.append(second).append(first).append("ma").append(rep);
            }

            if (i != words.length - 1) {
                converted.append(" ");
            }
        }

        return converted.toString();
    }

    public boolean isVowel(char c) {
        return c == 'a' || c == 'e' || c == 'u' || c == 'i' || c == 'o';
    }

    public String repeatChar(char c, int r) {
        StringBuilder result = new StringBuilder();

        for (int i = 0; i < r; i++) {
            result.append(c);
        }

        return result.toString();
    }
}

File: 0501-1000/0824_toGoatLatin.py
class Solution:
    def toGoatLatin(self, sentence: str) -> str:
        words = sentence.split()
        converted = ""

        for i, word in enumerate(words):
            if word[0] in ['a', 'e', 'i', 'o', 'u']:
                converted += word + 'ma' + 'a' * (i + 1)
            else:
                converted += word[1:] + word[0] + 'ma' + 'a' * (i + 1)
            if i != len(words) - 1:
                converted += " "
            
        return converted
